swarm.go

```diff
@@ -351,6 +358,13 @@ func (s *Swarm) NewStream(ctx context.Context, p peer.ID) (network.Stream, error
 				return nil, err
 			}
 		}
+
+		if c.Stat().Transient {
```

Comment:
can you add a unit test for this ?

Reply:
it would be nice, but this is quite hard to write and probably not worth the effort.